### ISSUE TITLE
Define & implement generic rank-to-rank lower comparison operator

### ISSUE THREAD

**Issue by ppebay** [DARMA-tasking/LB-analysis-framework#412]
PR #325 introduced the following less-than comparison operator of `Rank` instances:

```
    def __lt__(self, other):
        return self.get_load() < other.get_load()
```

However we do not necessarily want this to be always the comparison operator between ranks, as they can be sorted by other QoIs (like index, work, etc.). This issue must be discussed and only after consensus is reached, implemented more generically.

**Comment by ppebay** [comment]
No longer needed; replaced by different approach
